### ISSUE TITLE
error handling and performance

### ISSUE THREAD

**Issue by Byron** [Byron/godi#22]
Send godi through it's paces and manipulate an MHL seal file in every which way, verifying that errors reported will be meaningful (and actually are reported).

In a second step, profile it's memory usage and CPU usage, as previous tests have shown that it needs 3.5 times the CPU to increase overall speed by 2. It would be good to reduce overhead.

**Comment by Byron** [comment]
Error handling was greatly improved, as well as verify that now works per index-root (similar to what seal does).

Heap pressure was reduced, and buffer sizes adjusted to reduce syscall and communication overhead.

Now investigating one perfect way to increase performance, being to rethink the hashes it makes.
The one-in-a-universe chance of sha1 clashing is somewhat counteracted by an expensive MD5, which also is slower than SHA1. What if a super-cheap hash was made as backup instead for the non-mhl case ?

**Comment by Byron** [comment]
Turns out that MD5 is the fastest of the two, doing 600MB/s, whereas Sha1 'only' does 450MB/s. CRC64 clocks in at the same speed, whereas CRC32 only does 310MB/s.
In conclusion, MD5 is a perfectly valid backup hash, and not taking it wouldn't make anything faster in our case as we do it in parallel anyway.

It shows that simply not taking the MD5 as backup hash has a huge performance impact, bumping speed from 1.2GiB/s to 1.7GiB/s. Sha1 is the better hash of the two, maybe something like this could do it:

``` bash
godi --no-backup-hash seal A/
# Detects that MD5 is unset and doesn't do MD5 itself
godi verify A/godi.gobz
# Is forced to not do MD5
godi --no-backup-hash verify A/godi.gobz
```

As it has some implications, and performance is already 'rather good', I will take this into its own ticket.
